src/mlpack/core/data/one_hot_encoding_impl.hpp

```diff
@@ -56,14 +56,45 @@ void OneHotEncoding(const RowType& labelsIn,
     }
   }
   // Resize output matrix to necessary size, and fill it with zeros.
-  output.zeros(labelsIn.n_elem, curLabel);
+  output.zeros(curLabel, labelsIn.n_elem);
   // Fill ones in at the required places.
   for (size_t i = 0; i < labelsIn.n_elem; ++i)
   {
-    output(i, labels[i]) = 1;
+    output(labels[i], i) = 1;
   }
   labelMap.clear();
 }
+
+/**
+ * Overloaded function for the above function, which takes a matrix as input
+ * and also a vector of indices to encode and outputs a matrix.
+ *
+ * @param input Input dataset to be encoded.
+ * @param indices Index of rows to be encoded.
+ * @param output Encoded matrix.
+ */
+template<typename eT>
+void OneHotEncoding(arma::Mat<eT>& input,
+                    const arma::ucolvec indices,
+                    arma::Mat<eT>& output)
+{
+  output = input;
+  output.shed_rows(indices);
```

Comment:
It looks like shed_rows accepting a list of indices is pretty new, at least compared to the 8.400 requirment mlpack has. -- https://gitlab.com/conradsnicta/armadillo-code/-/commit/e6f5f536555c6314fdb574210c937634b6bf4e47

Reply:
Yes, IG that is why the build is failing, So now there are two possible solution:

Backport `shed_rows(indices)`

Or iterate through the list of indices and use `shed_row(uword index)`

I would prefer second one, Looking for inputs

Reply:
I agree; the second option seems cleaner.